server/grpc_service.go

```diff
@@ -396,7 +396,7 @@ func (s *Server) RegionHeartbeat(stream pdpb.PD_RegionHeartbeatServer) error {
 
 		region := core.RegionFromHeartbeat(request)
 		if region.GetLeader() == nil {
-			log.Error("invalid request, the leader is nil", zap.Reflect("reqeust", request), errs.ZapError(errs.ErrHeartBeatLeaderNil, err))
+			log.Error("invalid request, the leader is nil", zap.Reflect("reqeust", request), errs.ZapError(errs.ErrHeartBeatLeaderNil, nil))
```

Comment:
Would you like to change the `ZapError` to make the cause error as an optional parameter?